src/components/componentDiff/ComponentDiffTab.js

```diff
@@ -219,6 +221,79 @@ const ComponentVersionDiff = ({
     rightVersion: rightVersion,
   })
 
+  const determineMainSourceForComponent = (component) => {
```

Comment:
To make this function better reusable, rather let it return `null` over `<unknown>` and introduce `<unknown>` on consumer level. Also, you might add an `absentOk` flag which controls whether absence of main-source should be raised, or handled with `null`.